runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -283,6 +284,13 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
                 ch.pipeline().addLast(new NettyCorfuMessageEncoder());
                 ch.pipeline().addLast(new ClientHandshakeHandler(parameters.getClientId(),
                     node.getNodeId(), parameters.getHandshakeTimeout()));
+
+                // If parameters include message filters, add corresponding filter handler
+                if (parameters.getNettyClientInboundMsgFilters() != null) {
```

Comment:
We will have to discuss if this is also applicable for the Runtime running in the Management Agent on the server.

Reply:
These rules apply on any inbound messages received by client routers